File: algorithm/src/main/java/baekjoon/p10409/Main.java
package baekjoon.p10409;

import java.util.Scanner;

public class Main {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int t = sc.nextInt();
        int[] inputs = new int[n];

        for (int i = 0; i < inputs.length; i++) {
            inputs[i] = sc.nextInt();
        }

        System.out.println(solve(t, inputs));
    }

    private static int solve(int t, int[] inputs) {
        int sum = 0;
        int len = inputs.length;

        for (int i = 0; i < len; i++) {
            sum += inputs[i];
            if (sum > t) {
                return i;
            }
        }

        return len;
    }

}

File: algorithm/src/main/java/baekjoon/p10409/Main.kt
package baekjoon.p10409

fun main(args: Array<String>) {
    val (n, t) = readLine()!!.split(" ").map { it.toInt() }
    val inputs = readLine()!!.split(" ").map { it.toInt() }
    println(solve(t, inputs))
}

private fun solve(t: Int, inputs: List<Int>): Int {
    var sum = 0
    val size = inputs.size

    for (i in 0 until size) {
        sum += inputs[i]
        if (sum > t) {
            return i
        }
    }

    return size
}